src/abbrev.rs

```diff
@@ -0,0 +1,761 @@
+        }
+
+        Ok((input, abbrevs))
+    }
+}
+
+/// An abbreviation describes the shape of a `DebuggingInformationEntry`'s type:
+/// its code, tag type, whether it has children, and its set of attributes.
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct Abbreviation {
+    code: u64,
+    tag: constants::DwTag,
+    has_children: constants::DwChildren,
+    attributes: Vec<AttributeSpecification>,
+}
+
+impl Abbreviation {
+    /// Construct a new `Abbreviation`.
+    ///
+    /// ### Panics
+    ///
+    /// Panics if `code` is `0`.
+    pub fn new(code: u64,
+               tag: constants::DwTag,
+               has_children: constants::DwChildren,
+               attributes: Vec<AttributeSpecification>)
+               -> Abbreviation {
+        assert!(code != 0);
+        Abbreviation {
+            code: code,
+            tag: tag,
+            has_children: has_children,
+            attributes: attributes,
+        }
+    }
+
+    /// Get this abbreviation's code.
+    #[inline]
+    pub fn code(&self) -> u64 {
+        self.code
+    }
+
+    /// Get this abbreviation's tag.
+    #[inline]
+    pub fn tag(&self) -> constants::DwTag {
+        self.tag
+    }
+
+    /// Return true if this abbreviation's type has children, false otherwise.
+    #[inline]
+    pub fn has_children(&self) -> bool {
+        self.has_children == constants::DW_CHILDREN_yes
+    }
+
+    /// Get this abbreviation's attributes.
+    #[inline]
+    pub fn attributes(&self) -> &[AttributeSpecification] {
+        &self.attributes[..]
+    }
+
+    /// Parse an abbreviation's code.
+    fn parse_code(input: &[u8]) -> ParseResult<(&[u8], u64)> {
+        let (rest, code) = try!(parse_unsigned_leb(input));
+        if code == 0 {
+            Err(Error::AbbreviationCodeZero)
+        } else {
+            Ok((rest, code))
+        }
+    }
+
+    /// Parse an abbreviation's tag.
+    fn parse_tag(input: &[u8]) -> ParseResult<(&[u8], constants::DwTag)> {
+        let (rest, val) = try!(parse_unsigned_leb(input));
+        if val == 0 {
+            Err(Error::AbbreviationTagZero)
+        } else {
+            Ok((rest, constants::DwTag(val)))
+        }
+    }
+
+    /// Parse an abbreviation's "does the type have children?" byte.
+    fn parse_has_children(input: &[u8]) -> ParseResult<(&[u8], constants::DwChildren)> {
+        let (rest, val) = try!(parse_u8(input));
+        let val = constants::DwChildren(val);
+        if val == constants::DW_CHILDREN_no || val == constants::DW_CHILDREN_yes {
+            Ok((rest, val))
+        } else {
+            Err(Error::BadHasChildren)
+        }
+    }
+
+    /// Parse a series of attribute specifications, terminated by a null attribute
+    /// specification.
+    fn parse_attributes(mut input: &[u8]) -> ParseResult<(&[u8], Vec<AttributeSpecification>)> {
+        let mut attrs = Vec::new();
+
+        loop {
+            let result = AttributeSpecification::parse_null(input).map(|(rest, _)| (rest, None));
```

Comment:
Yeah, I wondered that too. I did attempt it for null abbreviations, but I don't think they have much impact on the current benchmark, so it was hard to tell and I left that for later.